vaadin-combo-box-flow/src/main/resources/META-INF/resources/frontend/comboBoxConnector.js

```diff
@@ -244,12 +244,26 @@ window.Vaadin.Flow.comboBoxConnector = {
 
       // Let server know we're done
       comboBox.$server.confirmUpdate(id);
+    }
+
+    customElements.whenDefined('vaadin-combo-box').then(() => {
+      const _isItemSelected = comboBox.$.overlay._isItemSelected;
+      // Override comboBox's _isItemSelected logic to handle remapped items
+      comboBox.$.overlay._isItemSelected = (item, selectedItem, itemIdPath) => {
+        let selected = _isItemSelected.call(comboBox, item, selectedItem, itemIdPath);
```

Comment:
Just a general comment: I personally think that `_isItemSelected.apply(comboBox, arguments)` is a cleaner method for monkey patching, as there's no need to duplicate all the parameters.